### ISSUE TITLE
Needs a README.md 

### ISSUE THREAD

**Issue by schnable** [democratic-csi/democratic-csi#4]
Something to say what it is and why I would want to use it.

**Comment by travisghansen** [comment]
Yes it does badly :( that’s why I never had it added to any kind of lists or ‘promoted’ it yet was missing doc.

In essence this project implements the `csi` spec for a storage provider for k8s. What that means is it makes persistent volume claims automatically provision storage, delete storage, expand storage, etc for your cluster workloads. For example say you need 10GB for a mysql DB...you ask the cluster for 10GB via a PVC and this tool will recognize that and automatically create the volume for you, setup iscsi (or nfs) shares and expose the storage, format the disk, etc so a pod can just use the storage. When you no longer need the storage you delete the PVC and this tool will reverse all the actions (remove the share, delete the volume, etc).

There are other csi implementations for other storage systems. This project specifically makes all the work if you have a FreeNAS storage appliance or if you ZFS on Linux with Ubuntu. In either case the outcome is the same, storage is managed by the software.

In the future I intend to add more drivers, specifically the ability to use zfs drives on the node for ephemeral volume support.

**Comment by schnable** [comment]
I am using the https://github.com/nmaupu/freenas-provisioner and saw you mention this project there.  I was interested in trying a CSI based interface and I saw you commenting that this might be close.  I have a raspberry pi cluster so I typically recompile stuff for arm (or arm64).    I think your overview above is a nice 1st cut.   It would be good to know about dependencies (specific version of K8s / freenas-provisioner?) and supported architectures too.  And the helm commands to deploy (and how to build from scratch).

**Comment by travisghansen** [comment]
Ah I see. The benefit over the old style provisioners is it supports all the new features none of which are being natively supported by non csi stuff (snapshots, resizing, ephemeral volumes, etc). The breadth of the spec is fully implemented here although the snapshots support (k8s support) was alpha when I built the helm charts so that needs to be updated as the move to beta was quite breaking.

- https://github.com/democratic-csi/charts/issues/5

Basically this driver itself is multiarch, however some of the k8s support tooling (that all csi drivers use) is not yet officially multiarch. There are several issues floating around to implement that in the various projects. In the mean time, the link above demonstrates someone who used some unofficial images to fill in the blanks and has it working on arm.

**Comment by travisghansen** [comment]
I think csi was available about 1.13..I was developing primarily against 1.15 with several feature gates enabled most of which are non longer necessary with latest k8s.

**Comment by schnable** [comment]
I might be able to provide a few cycles to help out - I can take a wack at creating a README if you like.    I will also give this a try on my home cluster (which is K8s 1.18.5 / arm).

**Comment by travisghansen** [comment]
Help is always welcome :)
